lib/rexml/parsers/baseparser.rb

```diff
@@ -259,7 +259,7 @@ def pull_event
           else
             @document_status = :after_doctype
             if @source.encoding == "UTF-8"
-              @source.buffer.force_encoding(::Encoding::UTF_8)
+              @source.buffer = @source.buffer.force_encoding(::Encoding::UTF_8)
```

Comment:
How about providing `Source#buffer_encoding=` instead of providing `Source#buffer=`?

Reply:
Instead of providing `Source#buffer=`, I would like to provide `Source#force_encoding`.

Reply:
I don't want to use `force_encoding` here because it doesn't change `REXML::Source#encoding` that is defined in `REXML::Encoding`. But `String#force_encoding` changes `String#encoding`. They will confuse us.

Reply:
I see.
Change to `Source#buffer_encoding=`.